Mathlib/CategoryTheory/ChosenFiniteProducts/Over.lean

```diff
@@ -0,0 +1,177 @@
+/-
+Copyright (c) 2025 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import Mathlib.CategoryTheory.ChosenFiniteProducts
+
+/-!
+
+# `ChosenFiniteProducts` for `Over X`
+
+We provide a `ChosenFiniteProducts (Over X)` instance via pullbacks, and provide simp lemmas
+for the induced `MonoidalCategory (Over X)` instance.
+
+-/
+
+namespace CategoryTheory.Over
+
+open Limits
+
+variable {C : Type*} [Category C] [HasPullbacks C]
+
+/-- A choice of finite products of `Over X` given by `Limits.pullback`. -/
+@[reducible]
+noncomputable
+def chosenFiniteProducts (X : C) : ChosenFiniteProducts (Over X) where
+  product Y Z := ⟨BinaryFan.mk (P := Over.mk (pullback.snd Y.hom Z.hom ≫ Z.hom))
+      (Over.homMk (pullback.fst Y.hom Z.hom) pullback.condition)
+      (Over.homMk (pullback.snd Y.hom Z.hom) rfl),
```

Comment:
We should have a definition saying that this binary fan IsLimit in Over X provided that the underlying square IsPullback, and then you can use that here. This specific product defined as a pullback shows up in elementary topos theory as well in a fundamental way. In fact `Mathlib.CategoryTheory.Limits.Constructions.Over.Products` gives the Has version of all this. Can you use / generalise that to give these? More specifically, I think we should have the IsBinaryProduct implies IsPullback versions, and everything else deduced from that.

Reply:
Ha, there's a prehistoric version of this here: https://github.com/b-mehta/topos/blob/c9032b11789e36038bc841a1e2b486972421b983/src/category/binary_products.lean#L271. I think that's what ended up being the construction I link above, but I'm not sure of the history. I have no doubt you can write something nicer :)

Reply:
Oops, you're right. I shouldn't be thinking about over categories at 2am, clearly!

Reply:
I should be more specific: the construction used in `over_binaryProduct_of_pullback` is the one I think should have an `Is` version (and currently doesn't, but the converse does), and that should be used to prove both `over_binaryProduct_of_pullback` and the new result here.

Reply:
Unfortunately the generic abstract nonsense doesn't give the right defeqs, but I've managed to add the construction by hand.